src/mono/wasm/build/WasmApp.targets

```diff
@@ -115,6 +116,8 @@
     <WasmEnableExceptionHandling Condition="'$(WasmEnableExceptionHandling)' == ''">true</WasmEnableExceptionHandling>
     <WasmEnableSIMD Condition="'$(WasmEnableSIMD)' == ''">$(WasmEnableExceptionHandling)</WasmEnableSIMD>
     <WasmEnableLegacyJsInterop Condition="'$(WasmEnableLegacyJsInterop)' == ''">true</WasmEnableLegacyJsInterop>
+    <WasmEnableJsInteropByValue Condition="'$(WasmEnableJsInteropByValue)' == '' and ( '$(WasmEnableThreads)' == 'true' or '$(MonoWasmBuildVariant)' == 'multithread' )">true</WasmEnableJsInteropByValue>
```

Comment:
Isn't `WasmEnableThreads` user facing, and `MonoWasmBuildVariant` is meant more for the wasm/runtime build itself? I see that it is being used in a couple of other places, which we should fix too. Not necessary for this PR.

Reply:
This makes it easy to work in-tree with samples and unit tests when you just have `MonoWasmBuildVariant` set globally and sometimes native rebuild kicks it. Maybe it could be unified somehow, I'm not 100% sure about it.

Reply:
You can set `WasmEnableThreads` based on `MonoWasmBuildVariant` in `WasmApp.InTree.targets`.

Reply:
I would prefer if you could help me to improve it in different PR. But I would like to ask details in meantime

what are you supposed to use `WasmEnableThreads` or `MonoWasmBuildVariant` inside of
- wasm.proj
- WasmApp.Native.targets 
- C code like drive.c triggered by runtime build or application native rebuild
- inside .csproj of libraries.
- inside .csproj of libraries unit tests
- inside .csproj of in-tree samples
- in .csproj of WBT

What are you supposed to use `WasmEnableThreads` or `MonoWasmBuildVariant` on commandline ?
- when you build runtime and libraries with `build.sh`
- when you run unit tests. This could trigger re-build of C# libraries
- when you run samples in-tree.  This could trigger re-build of C# libraries

How does it propagate to the nested build/publish ?
Does build on helix changes anything about it ?
Does WBT runtime installation in artifacts changes anything about it ?
Is there really benefit of having 2 of them ?

Reply:
`MonoWasmBuildVariant` is really only for non-user projects, so wasm.proj would be an example.
All the other test cases (except WBT) use InTree targets, so adding overrides there would apply to all of them.
WBT strictly uses what the user gets, so no InTree targets.

> Is there really benefit of having 2 of them ?

Not really. I think we had that before for `multithread`, and `perftrace` builds but now we don't need a string property. It is clearer to have them separate but I wouldn't mind using only `WasmEnableThreads` either.
What I do want to avoid is using internal build properties in user facing targets.

cc @lambdageek